src/deno/_pointers.ts

```diff
@@ -32,6 +46,37 @@ export function denoFromPlatformPointer<T>(value: PlatformPointer<T>): Pointer<T
   return new Pointer(value as unknown as PlatformPointer<T>);
 }
 
+export function denoToPlatformStruct(data: TypedArray): Uint8Array;
+export function denoToPlatformStruct<T extends AllocatableStruct>(
+  data: Pointer<T>,
+  dataType: AllocatableStructConstructor<T>,
+): Uint8Array;
+export function denoToPlatformStruct<T extends AllocatableStruct>(
+  data: TypedArray | Pointer<T>,
+  dataType: AllocatableStructConstructor<T>,
+): Uint8Array;
+export function denoToPlatformStruct<T extends AllocatableStruct>(
+  data: TypedArray | Pointer<T>,
+  dataType?: AllocatableStructConstructor<T>,
+): Uint8Array {
+  if (data instanceof Uint8Array) {
+    return data;
+  } else if (isTypedArray(data)) {
+    return new Uint8Array(data.buffer);
+  } else {
+    if (isTypedArray(data._data)) {
+      return new Uint8Array(data._data.buffer);
+    } else {
+      const view = new DenoPlatformDataView(data._data);
```

Comment:
Suggestion: Let's not build the `view` until after we've checked `dataType`. 

```suggestion
      if (dataType === undefined) {
        throwError("Cannot retrieve a buffer from a pointer without type!");
      }

      const view = new DenoPlatformDataView(data._data);
      return view.getArray(dataType.SIZE_IN_BYTES, 0);
```